assets/js/blocks/mini-cart/edit.tsx

```diff
@@ -59,6 +92,47 @@ const Edit = ( { attributes, setAttributes }: Props ): ReactElement => {
 	const productCount = 0;
 	const productTotal = 0;
 
+	const colorGradientSettings = useMultipleOriginColorsAndGradients();
+
+	const colorSettings = [
+		{
+			value: priceColorValue,
+			onChange: ( colorValue: string ) => {
+				setPriceColor( colorValue );
+				setAttributes( { priceColorValue: colorValue } );
```

Comment:
I'm curious why this is calling both `setPriceColor` and `setAttributes` for the color? The `setPriceColor` setter comes from using the `withColors` HOC and looks like internally it already calls `setAttributes`?

https://github.com/WordPress/gutenberg/blob/c5d477ab54ad98d3998809ba1e72675eb70f48bc/packages/block-editor/src/components/colors/with-colors.js#L136C14-L145

Reply:
I have followed the example of the https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/social-links/edit.js implementation which does it that way.

I have tested the code without `setAttributes` and it seems not to be working.

Reply:
Probably not working because what you set in as the object for `withColors` is `priceColor` whereas the registered context is consuming `priceColorValue` so to update the context you need to attach the color value to `priceColorValue`.  I might be missing something, but why is `priceColor` being used as the prop sent into `withColors` and `priceColorValue` used for the context? I understand you're following the example of social icons but I wonder if there is something that could be done better there too.